db/schema/parallel/nft_royalty.go

```diff
@@ -0,0 +1,105 @@
+package parallel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+	"github.com/likecoin/likecoin-chain-tx-indexer/utils"
+)
+
+func MigrateNftRoyalty(conn *pgxpool.Conn, batchSize uint64) error {
+	err := checkBatchSize(batchSize)
+	if err != nil {
+		return err
+	}
+	err = checkMinSchemaVersion(conn, 13)
+	if err != nil {
+		return err
+	}
+	return conn.BeginFunc(context.Background(), func(dbTx pgx.Tx) error {
+		_, err := dbTx.Exec(context.Background(), `
+			DECLARE nft_royalty_migration_cursor CURSOR FOR
+				SELECT id, class_id, nft_id, tx_hash, events
+				FROM nft_event
+				ORDER BY id
+			;
+		`)
+		if err != nil {
+			logger.L.Errorw("Error when declaring cursor", "error", err)
+			return err
+		}
+
+		for {
+			rows, err := dbTx.Query(context.Background(), fmt.Sprintf(`FETCH %d FROM nft_royalty_migration_cursor`, batchSize))
+			if err != nil {
+				logger.L.Errorw("Error when fetching from cursor", "error", err)
+				return err
+			}
+
+			if rows == nil {
+				break
+			}
+
+			var pkeyId int64
+			var royalties []db.NftRoyalty
+			for rows.Next() {
+				var classId string
+				var nftId string
+				var txHash string
+				var eventRaw []string
+				err = rows.Scan(&pkeyId, &classId, &nftId, &txHash, &eventRaw)
+				if err != nil {
+					logger.L.Errorw("Error when scanning row", "error", err)
+					return err
+				}
+
+				events, err := utils.ParseEvents(eventRaw)
+				if err != nil {
+					logger.L.Errorw("Error when parsing events", "error", err)
+					return err
+				}
+
+				msgAction := utils.GetEventsValue(events, "message", "action")
+				if msgAction == "/cosmos.bank.v1beta1.MsgSend" || msgAction == string(db.ACTION_BUY) || msgAction == string(db.ACTION_SELL) {
```

Comment:
Should use `db.ACTION_SEND`

Reply:
Does `db.ACTION_SEND` refer to `/cosmos.nft.v1beta1.MsgSend`? 
I think the events of `ACTION_SEND` in `logs` array of a tx ([sample](https://mainnet-node.like.co/cosmos/tx/v1beta1/txs/DF3D66EC7C9EAE28EE076D4FC80268FA42595957BC323B9C48A879EB453BDA22)) is as the below format:
```
{
  "msg_index": 1,
  "log": "",
  "events": [
    {
      "type": "cosmos.nft.v1beta1.EventSend",
      "attributes": [
        {
          "key": "sender",
          "value": "\"like17m4vwrnhjmd20uu7tst7nv0kap6ee7js69jfrs\""
        },
        {
          "key": "receiver",
          "value": "\"like1a33400g7wrlv04tzdjzqwpar5rs4qhkakz8y3w\""
        },
        {
          "key": "class_id",
          "value": "\"likenft1t5ftaxzyrnafewsav8qxvvsafs59a8xe2wnrl594yag04u9zwsjqrq0vgt\""
        },
        {
          "key": "id",
          "value": "\"writing-1a4bc1b2-fe70-4033-9c76-1424b66d4f16\""
        }
      ]
    },
    {
      "type": "message",
      "attributes": [
        {
          "key": "action",
          "value": "/cosmos.nft.v1beta1.MsgSend"
        }
      ]
    }
  ]
},
```
Should we instead fetch the `coin_received` event attributes from `/cosmos.bank.v1beta1.MsgSend`?

Reply:
Oh I missed the module name, I thought it was `/cosmos.nft.v1beta1.MsgSend`